### PR TITLE
Readme updates: add links, capitalization, & formatting

### PR DESCRIPTION
**Pull request by gwerbin**
These changes are intended to make the Readme easier to understand.

I have emphasized the difference between PyPy and CPython.

Adding links will also reduce confusion if a user does not know what e.g. PyPy and Mamba are. Mamba in particular is difficult to search for, because it shares a name with at least 2 other established Python-related projects.

_I have omitted the checklist because this PR is related to the Readme and not to any feedstocks._

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
-# miniforge
-![Build miniforge](https://github.com/conda-forge/miniforge/workflows/Build%20miniforge/badge.svg)
+# Miniforge
+![Build Miniforge](https://github.com/conda-forge/miniforge/workflows/Build%20miniforge/badge.svg)
 
-This repository holds a minimal installer for conda specific to conda-forge. It is comparable to [Miniconda](https://docs.conda.io/en/latest/miniconda.html), but with
+This repository holds a minimal installer for [Conda](https://conda.io/) specific to [Conda Forge](https://conda-forge.org/). It is comparable to [Miniconda](https://docs.conda.io/en/latest/miniconda.html), but with:
```
Comment (hmaarrfk) [inline README.md:4]:
```suggestion
This repository holds a minimal installer for [Conda](https://conda.io/) specific to [conda-forge](https://conda-forge.org/). It is comparable to [Miniconda](https://docs.conda.io/en/latest/miniconda.html), but with:
```

I'm not sure iv'e seen conda-forge capitalized ever.

Comment (gwerbin) [reply README.md]: IMO it should be, but I'm not about to press that issue here :P

File: README.md [hunk 2]
```diff
@@ -23,13 +25,13 @@ Latest installers with python 3.8 `(*)` in the base environment `(**)`
 | OS X    | arm64 (Apple Silicon) | [Miniforge3-MacOSX-arm64](https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh) |
 | Windows | x86_64                | [Miniforge3-Windows-x86_64](https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Windows-x86_64.exe) |
 
-`(*)` OS X arm64 will be installed with Python 3.9
+`(*)` OS X `arm64` will be installed with Python 3.9.
 
-`(**)` the python version is specific only to the base environment. installed conda can create new environments with different python versions and implementations
+`(**)` the Python version is specific only to the base environment. Conda/Mamba can create new environments with different Python versions and implementations.
```
Comment (hmaarrfk) [inline README.md:30]:
```suggestion
`(**)` the Python version is specific only to the base environment. Conda can create new environments with different Python versions and implementations.
```

**Comment by hmaarrfk** [comment]
Thanks!

**Pull request merged**